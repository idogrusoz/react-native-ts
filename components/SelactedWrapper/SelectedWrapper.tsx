import React from "react";
import { TouchableOpacity, StyleSheet } from "react-native";
import { useDispatch, useSelector } from "react-redux";
import { Text } from "../Themed";
import Selected from "../Selected/Selected";
import { primary } from "../../constants/Colors";
import { ingredientsSelector } from "../../redux/selectors";
import { clearSelected, setLoading, setSearchResults } from "../../redux/actions/actions";
import { Ingredient } from "../../types";
import { useNavigation } from "@react-navigation/native";

const SelectedWrapper = () => {
    const { selected } = useSelector(ingredientsSelector);
    const dispatch = useDispatch();
    const navigation = useNavigation();
    const handleSearch = () => {
        dispatch(setLoading(true));
        dispatch(setSearchResults(selected.map((item: Ingredient) => item.name)));
        navigation.navigate("SearchResults");
        dispatch(clearSelected());
    };
    return (
        <>
            {selected.map((item, i) => {
                return <Selected name={item.name} index={i} key={item.id.toString()} />;
            })}
            {selected.length > 0 && (
                <TouchableOpacity onPress={handleSearch} style={styles.button}>
                    <Text style={styles.buttonLabel}>Search</Text>
                </TouchableOpacity>
            )}
        </>
    );
};

const styles = StyleSheet.create({
    button: {
        display: "flex",
        justifyContent: "center",
        alignItems: "center",
        borderRadius: 5,
        color: "#fff",
        backgroundColor: primary,
        height: 40,
        width: 90,
        alignSelf: "center",
        marginVertical: 15,
    },
    buttonLabel: {
        color: "#fff",
    },
});

export default SelectedWrapper;
