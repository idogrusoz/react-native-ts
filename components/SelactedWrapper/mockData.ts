import { Ingredient } from "../../types";

const mockIngredient1: Ingredient = {
    id: 1,
    name: "name1",
    image: "image1",
};
const mockIngredient2: Ingredient = {
    id: 2,
    name: "name2",
    image: "image2",
};
export const mockSelected: Ingredient[] = [mockIngredient1, mockIngredient2];
