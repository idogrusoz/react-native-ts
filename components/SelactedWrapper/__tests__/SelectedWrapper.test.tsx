import React from "react";
import { render, RenderAPI } from "@testing-library/react-native";
import SelectedWrapper from "../SelectedWrapper";
import * as redux from "react-redux";
import { mockSelected } from "../mockData";
const mockedNavigate = jest.fn();
jest.mock("@react-navigation/native", () => {
    return {
        useNavigation: () => ({
            navigate: mockedNavigate,
        }),
    };
});
const spySelect = jest.spyOn(redux, "useSelector");
const spyDispatch = jest.spyOn(redux, "useDispatch");

describe("SelectedWrapper test", () => {
    let wrapper: RenderAPI;

    beforeEach(() => {
        spySelect.mockReturnValue({ selected: mockSelected });
        spyDispatch.mockImplementation(jest.fn());
        wrapper = render(<SelectedWrapper />);
    });
    it("renders without crash", () => {
        expect(wrapper).toMatchSnapshot();
    });
});
