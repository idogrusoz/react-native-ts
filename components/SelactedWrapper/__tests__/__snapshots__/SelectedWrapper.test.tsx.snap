// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedWrapper test renders without crash 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "height": 25,
          "justifyContent": "space-between",
          "marginVertical": 10,
          "paddingHorizontal": 30,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
          },
          Object {
            "fontFamily": "System",
          },
          undefined,
        ]
      }
    >
      name1
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "height": 20,
          "justifyContent": "center",
          "opacity": 1,
          "width": 30,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
            },
            Object {
              "fontFamily": "System",
            },
            undefined,
          ]
        }
      >
        x
      </Text>
    </View>
  </View>,
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "height": 25,
          "justifyContent": "space-between",
          "marginVertical": 10,
          "paddingHorizontal": 30,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
          },
          Object {
            "fontFamily": "System",
          },
          undefined,
        ]
      }
    >
      name2
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "height": 20,
          "justifyContent": "center",
          "opacity": 1,
          "width": 30,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
            },
            Object {
              "fontFamily": "System",
            },
            undefined,
          ]
        }
      >
        x
      </Text>
    </View>
  </View>,
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#00701a",
        "borderRadius": 5,
        "color": "#fff",
        "display": "flex",
        "height": 40,
        "justifyContent": "center",
        "marginVertical": 15,
        "opacity": 1,
        "width": 90,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
          },
          Object {
            "fontFamily": "System",
          },
          undefined,
        ]
      }
    >
      Search
    </Text>
  </View>,
]
`;
